### PR TITLE
Update documentation and convert to Markdown

### PR DESCRIPTION
**Pull request by jeremybennett**
Update main README.md to match new structure of repository and
	scripts.  Convert user guide from Asciidoc to Markdown and rewrite
	for new Python scripts.  Add a script to generate a table of
	contents and a Makefile for convenience.

Files changed:

	* README.md: Updated to match current repository structure
	* doc/Makefile: Created.
	* doc/README.adoc: Deleted.
	* doc/README.md: Created by converting README.adoc to Markdown.
	* doc/gentoc.sh: Created.

### CODE REVIEW

File: doc/README.md [hunk 1]
```diff
@@ -0,0 +1,1035 @@
...
+
+These are combined into the general header `support.h` which is included by
+all benchmarks, and defines values which may be used by the benchmarks. The
+values that may be defined are:
+
+Board specific code that is to be linked in to the benchmarks should be
+defined in `config/ARCH/boards/BOARDNAME/boardsupport.c`.  This file should
+define the following functions:
+
+- `initialize_board` which is alled to initialize the board;
```
Comment (kohnakagawa) [inline doc/README.md]: spelling typo

alled -> called

File: README.md [hunk 2]
```diff
@@ -13,7 +13,8 @@ and Trevor Mudge (see https://tmt.knect365.com/risc-v-workshop-zurich/agenda/2#s
 
 The benchmarks are largely derived from the Bristol/Embecosm Embedded
 Benchmark Suite (BEEBS, see http://beebs.eu), which in turn draws its material
-from various earlier projects.  A full description and user manual is in the [`doc` directory](./doc/README.adoc).
+from various earlier projects.  A full description and user manual is in the
+[`doc` directory](./doc/README.adoc).
```
[the remark this answers is not in the document]
Comment (jeremybennett) [reply README.md]: Thanks - corrected.

File: doc/README.md (hunk 1 above)
Comment (jeremybennett) [reply doc/README.md]: Thanks - corrected.

**Comment by jeremybennett** [comment]
@kohnakagawa, @oferShinaar Thanks for your comments. Corrected documentation merged into master

**Pull request merged**